### PR TITLE
test: add URL construction helpers

### PR DESCRIPTION
**Pull request by jpeach**
The integration testsuite uses passed-in parameters to construct a
URL that can be used with the target load balancer. Add a `url` helper
package to use those parameters to construct URL strings so that the
tests themselves can be eadier to read and less fiddly.

Signed-off-by: James Peach <jpeach@vmware.com>

### CODE REVIEW

File: _integration/testsuite/policies/contour-client.rego [hunk 1]
```diff
@@ -14,45 +14,35 @@
...
 # send requests through.
 target_http_port = port {
-  port := data.test.params.proxy.http_port
-}
-
-# target_https_port returns the TLS port for the proxy that tests should
-# send requests through.
-target_https_port = "9143" {
-  not data.test.params.proxy.https_port
+  trace(sprintf("^^^^ port=%s", [data.test.params.proxy.http_port]))
```
Comment (stevesloka) [inline _integration/testsuite/policies/contour-client.rego:28]: debug?

File: _integration/testsuite/policies/contour-url.rego [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+package contour.http.client.url
+
+import data.contour.http.client
+
+# Generate a HTTP URL from the `target_addr` and `target_http_port`
+# parameters.  The path argument specifies the full URL path and must
+# include the leading '/'.
+http(path) = url {
+    url := sprintf("http://%s:%d%s", [
```
Comment (stevesloka) [inline _integration/testsuite/policies/contour-url.rego:23]: Should we have a check that the path begins with a `/`?

File: _integration/testsuite/policies/contour-client.rego (hunk 1 above)
Comment (jpeach) [reply _integration/testsuite/policies/contour-client.rego:28]: Oops, good catch!

File: _integration/testsuite/policies/contour-url.rego (hunk 2 above)
Comment (jpeach) [reply _integration/testsuite/policies/contour-url.rego:23]: Yeh, I got a bit lazy here, but you are right that we can do a bit better here. I'll look into adding the leading '/' only if it is necessary.

Comment (stevesloka) [reply _integration/testsuite/policies/contour-url.rego:23]: I guess it's not dynamic, so it shouldn't really come up. ¯\_(ツ)_/¯

File: _integration/testsuite/policies/contour-client.rego (hunk 1 above)
Comment (jpeach) [reply _integration/testsuite/policies/contour-client.rego]: Fixed.

File: _integration/testsuite/policies/contour-url.rego (hunk 2 above)
Comment (jpeach) [reply _integration/testsuite/policies/contour-url.rego:23]: Fixed.

**Pull request merged**